nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/cf/JndiJmsConnectionFactoryProperties.java

```diff
@@ -114,4 +119,67 @@ public static PropertyDescriptor getDynamicPropertyDescriptor(final String prope
                 .build();
     }
 
+    private static class JndiJmsContextFactoryValidator implements Validator {
+        private static final String DISALLOWED_CONTEXT_FACTORY = "LdapCtxFactory";
+
+        @Override
+        public ValidationResult validate(final String subject, final String input, final ValidationContext context) {
+            final ValidationResult.Builder builder = new ValidationResult.Builder().subject(subject).input(input);
+
+            if (input == null || input.isEmpty()) {
+                builder.valid(false);
+                builder.explanation("Context Factory is required");
+            } else if (input.endsWith(DISALLOWED_CONTEXT_FACTORY)) {
+                builder.valid(false);
+                builder.explanation(String.format("Context Factory [%s] not allowed", DISALLOWED_CONTEXT_FACTORY));
+            } else {
+                builder.valid(true);
+                builder.explanation("Context Factory allowed");
+            }
+
+            return builder.build();
+        }
+    }
+
+    private static class JndiJmsProviderUrlValidator implements Validator {
+        /** JNDI JMS URL Allowed Schemes based on ActiveMQ Connection Factory */
+        private static final Set<String> ALLOWED_SCHEMES = Collections.unmodifiableSet(new LinkedHashSet<>(Arrays.asList(
+                "jgroups",
+                "tcp",
+                "udp",
+                "vm"
+        )));
```

Comment:
@exceptionfactory I have concerns that we can enumerate all supported schemes because they are vendor specific. E.g. WebLogic uses `t3(s)://`. Basically, it can be an arbitrary name.

I also understand that `ldap://` is problematic but WebSphere MQ relies on it when using JNDI (or the local `file://` can be used).

Do you have any idea how to handle these shortcomings while keeping the validation logic?

Reply:
Thanks for highlighting these concerns @turcsanyip.

The restrictive approach to URL schemes does present some potential limitations.

I expanded the default list of allowed schemes, and I also introduced a Java System property that supports overriding the default values. I updated the additional details documentation to list the property name and describe how it can be configured. This approach will allow an administrator to customize validation for specific environments and custom drivers while providing default validation for general deployments.